devimint/src/external.rs

```diff
@@ -240,13 +241,54 @@ impl Bitcoind {
         Ok(proof.encode_hex())
     }
 
-    pub async fn get_raw_transaction(&self, txid: bitcoin30::Txid) -> Result<String> {
+    /// Get a transaction by its txid. Checks the mempool and all blocks.
+    pub async fn get_transaction(&self, txid: bitcoin30::Txid) -> Result<String> {
+        // Check the mempool.
+        if let Ok(tx) = self.get_raw_transaction(txid, None).await {
+            return Ok(tx);
+        };
+
+        let block_height = self.get_block_count().await? - 1;
+
+        // Check each block for the tx, starting at the chain tip.
+        // Buffer the requests to avoid spamming bitcoind.
+        // We're doing this after checking the mempool since the tx should
+        // usually be in the mempool, and we don't want to needlessly hit
+        // the bitcoind with block requests.
+        let mut buffered_tx_stream = futures::stream::iter((0..block_height).rev())
+            .map(|height| async move {
+                let block_hash = self.get_block_hash(height).await?;
+                let tx = self.get_raw_transaction(txid, Some(block_hash)).await?;
+                Ok::<_, anyhow::Error>(tx)
```

Comment:
Can we distinguish `Ok(None)` from `Err(_)` somehow? Seems to me that we should return on error (we expect `bitcoind` to respond correctly) and just continue scanning if we just couldn't find the tx.

Reply:
Good point! Done. It's a bit hacky, but I dug into the bitcoind source code and figured out that a JSON-RPC response with an error code of -5 indicates that the transaction doesn't exist. I linked to the relevant code in this PR.